semana7/spotif4/src/components/ShowPlaylist/index.js

```diff
@@ -0,0 +1,181 @@
+import React, { Component } from 'react';
+import * as S from './styled';
+import axios from 'axios';
+
+const baseUrl = 'https://us-central1-spotif4.cloudfunctions.net/api';
+
+class ShowPlaylist extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      playlist: [],
+      name: '',
+      artist: '',
+      url: ''
+    };
+  }
+
+  componentDidMount() {
+    this.getPlaylistMusics(this.props.playlistId);
+  }
+
+  getPlaylistMusics = async id => {
+    try {
+      const response = await axios.get(
+        `${baseUrl}/playlists/getPlaylistMusics/${id}`,
+        {
+          headers: {
+            auth: 'string'
+          }
+        }
+      );
+      this.setState({ playlist: response.data.result.musics });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  addMusicToPlaylist = async (name, artist, url) => {
+    try {
+      await axios.put(
+        `${baseUrl}/playlists/addMusicToPlaylist`,
+        {
+          playlistId: this.props.playlistId,
+          music: {
+            name: name,
+            artist: artist,
+            url: url
+          }
+        },
+        {
+          headers: {
+            auth: 'string'
+          }
+        }
+      );
+      this.setState({ name: '', artist: '', url: '' });
+      this.getPlaylistMusics(this.props.playlistId);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  removeMusicFromPlaylist = async (playlistId, musicId) => {
+    const deleteMusic = window.confirm(
+      'Tem certeza de que deseja remover essa música da playlist?'
+    );
+
+    if (deleteMusic) {
+      try {
+        await axios.delete(
+          `${baseUrl}/playlists/removeMusicFromPlaylist?playlistId=${playlistId}&musicId=${musicId}`,
+          {
+            headers: {
+              auth: 'string'
+            }
+          }
+        );
+        alert('A música foi removida com sucesso!');
+        this.getPlaylistMusics(this.props.playlistId);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  };
+
+  getNameInput = event => {
+    this.setState({ name: event.target.value });
+  };
+
+  getArtistInput = event => {
+    this.setState({ artist: event.target.value });
+  };
+
+  getUrlInput = event => {
+    this.setState({ url: event.target.value });
+  };
+
+  render() {
+    const showPlaylistMusics = this.state.playlist.map((el, index) => {
+      console.log(el.url);
+      return (
+        <tr key={index}>
+          <td>{el.name}</td>
+          <td>{el.artist}</td>
+          <td>
+            <audio controls="controls">
+              <source src={el.url} type="audio/mp3" />
+            </audio>
+          </td>
+          <td>
+            <S.Action
+              bgred
+              onClick={() => {
+                this.removeMusicFromPlaylist(this.props.playlistId, el.id);
+              }}
```

Comment:
Mesma coisa que comentei lá embaixo, mas com um adendo, você não precisa passar via parâmetros algo que o método tem acesso direto, ou seja, de dentro do seu método `removeMusicFromPlaylist ` você já consegue ter acesso ao `this.props.playlistId`, então passá-lo como parâmetro é desnecessário.